packages/zoltan2/src/directory/Zoltan2_Directory_Impl.hpp

```diff
@@ -901,13 +842,11 @@ int Zoltan2_Directory<gid_t,lid_t,user_t>::find_local(
   // and use index to call value_at. Alternative is to call exists(*gid) and
   // then node_map.value_at(node_map.find(*gid))) which is slower.
   // TODO: Can this be optimized further?
-  // size_t node_index = node_map.find(*gid); // TODO: Switch to Kokkos Map
-  if(node_map.find(*gid) != node_map.end())
-  // if(node_map.valid_at(node_index))
+  size_t node_index = node_map.find(*gid);
+  if(node_map.valid_at(node_index))
   {
     const Zoltan2_Directory_Node<gid_t,lid_t,user_t> & node =
-      node_map.at(*gid);
-    //  node_map.value_at(node_index); // TODO: Switch to Kokkos map
+      node_map.value_at(node_index);
     /* matching global ID found! Return gid's information */
     if(lid) {
```

Comment:
Not a fan of a pointer being called `lid`.  That confused me until I saw the next line.

Reply:
Agreed. Some code is still replicated from zoltan and as I continue to work on it, I need to convert the original c code to be more c++ like. We plan to return to this to add parallel forms. I will improve it.